roles/migrationcontroller/tasks/main.yml

```diff
@@ -371,16 +366,91 @@
           state: present
           definition: "{{ lookup('file', '/tmp/authentication.yaml') | from_yaml }}"
 
+- name: "Set up mig ui"
+  k8s:
+    state: "{{ ui_state }}"
+    definition: "{{ lookup('template', 'ui.yml.j2') }}"
+
 - when: mig_svc_account
   block:
   - name: "Set up mig svc account for remote connections"
     k8s:
       state: "present"
       definition: "{{ lookup('template', 'mig_service_account.yml.j2') }}"
     when: not olm_managed
-  
+
 - when: migration_controller or migration_ui
   name: "Set up host MigCluster"
   k8s:
     state: "present"
     definition: "{{ lookup('template', 'mig_host_cluster.yml.j2') }}"
+
+- when: not(migration_controller)
+  block:
+  - name: Find Controller ReplicaSets
+    k8s_facts:
+      kind: ReplicaSet
+      namespace: "{{ mig_namespace }}"
+      label_selectors:
+      - control-plane=controller-manager
+    register: controller_replicasets
+
+  - name: Terminate Controller ReplicaSets
+    k8s:
+      name: "{{ item.metadata.name }}"
+      state: absent
+      kind: ReplicaSet
+      namespace: "{{ mig_namespace }}"
+    with_items: "{{ controller_replicasets.resources }}"
+
+- when: not(migration_ui)
+  block:
+  - name: Find UI ReplicaSets
+    k8s_facts:
+      kind: ReplicaSet
+      namespace: "{{ mig_namespace }}"
+      label_selectors:
+      - service=migration-ui
+    register: ui_replicasets
+
+  - name: Terminate UI ReplicaSets
+    k8s:
+      name: "{{ item.metadata.name }}"
+      state: absent
+      kind: ReplicaSet
+      namespace: "{{ mig_namespace }}"
+    with_items: "{{ ui_replicasets.resources }}"
+
+- when: not(migration_velero)
+  block:
+  - name: Find Velero ReplicaSets
+    k8s_facts:
+      kind: ReplicaSet
+      namespace: "{{ mig_namespace }}"
+      label_selectors:
+      - component=velero
+    register: velero_replicasets
+
+  - name: Terminate Velero ReplicaSets
+    k8s:
+      name: "{{ item.metadata.name }}"
+      state: absent
+      kind: ReplicaSet
+      namespace: "{{ mig_namespace }}"
+    with_items: "{{ velero_replicasets.resources }}"
+
+  - name: Find Restic Pods
+    k8s_facts:
+      kind: Pod
+      namespace: "{{ mig_namespace }}"
+      label_selectors:
+      - name=restic
+    register: restic_pods
+
+  - name: Terminate Restic Pods
+    k8s:
+      name: "{{ item.metadata.name }}"
+      state: absent
+      kind: Pod
+      namespace: "{{ mig_namespace }}"
+    with_items: "{{ restic_pods.resources }}"
```

Comment:
Removing Deployments does not remove Replicasets. Removing Daemonsets does not remove pods.

These blocks will do the needful to ensure all the pods are shut down.